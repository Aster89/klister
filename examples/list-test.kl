#lang "prelude.kl"

-- These examples are here rather than in "list.kl" so that the resulting
-- values are tracked by our test suite, which only covers the "examples"
-- folder, not "stdlib".

(import "list.kl")

(example
  (foldr + 0 (list 1 2 3 4)))

(example
  (map string-length (list "foo" "bar" "baz" "quux")))

(example
  (filter (lambda (x) (< x 10))
    (list 1 11 111 2 22 222 3 33 333)))

(example
  (snoc (list 1 2 3) 4))

(example
  (reverse (list 1 2 3)))

(example
  (syntax->list '(a b c)))

(example
  (list->syntax (list 'a 'b 'c) 'here))
