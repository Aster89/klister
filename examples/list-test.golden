10 : Integer
(:: 3 (:: 3 (:: 3 (:: 4 (nil))))) : (List Integer)
(:: 1 (:: 2 (:: 3 (nil)))) : (List Integer)
(:: 1 (:: 2 (:: 3 (:: 4 (nil))))) : (List Integer)
(:: 3 (:: 2 (:: 1 (nil)))) : (List Integer)
(:: #[list-test.kl:26.19-26.20]<a>
    (:: #[list-test.kl:26.21-26.22]<b>
        (:: #[list-test.kl:26.23-26.24]<c> (nil)))) : (List Syntax)
#[list-test.kl:29.34-29.38]<(a b c)> : Syntax
